event_record.rb

```diff
@@ -0,0 +1,145 @@
+require 'date'
+require_relative 'event_details'
+class EventRecord
+  @@events_date = Hash.new{|hash, key| hash[key] = []}
+  @@events_details = {}
+  @@id = 0
+
+  def add_event_details(name, date)
+    begin
+      date = Date.parse(date)
+      @@id += 1
+      @@events_details[@@id] = Event_Details.new(@@id, name, date)
+      year_month_key = "#{date.year.to_s}-#{date.month.to_s}"
+      day = date.day
+      @@events_date[year_month_key].push([name, day])
+      puts "Event added"
+      rescue
+        puts "Incorrect Data"
+    end
+  end
+
+  def self.return_all_events
+    if @@events_details.length == 0
```

Comment:
write success scenario first and then blank scenario.

Reply:
done